### ISSUE TITLE
[TW-918] crash when running 'task log' with a project

### ISSUE THREAD

**Issue by taskwarrior** [GothenburgBitFactory/taskwarrior#953]
_Itay Perl on 2010-11-20T06:29:49Z says:_ 

 The following command would crash on v1.9.3: task log foo pro:bar.

<pre>
1. 0  fseek (fp=0x0, offset=0, whence=0) at fseek.c:40
1. 1  0x0809c095 in TDB::loadCompleted (this=0x817d988, tasks=..., filter=...) at TDB.cpp:394
1. 2  0x080a13a4 in TDB::load (this=0x817d988, tasks=..., filter=...) at TDB.cpp:271
1. 3  0x081206b1 in onProjectChange (task=..., scope=true) at report.cpp:2884
1. 4  0x080c753d in handleLog (outs=...) at command.cpp:203
1. 5  0x0806bed1 in Context::dispatch (this=0x817d920, out=...) at Context.cpp:225
1. 6  0x0806c77d in Context::run (this=0x817d920) at Context.cpp:144
1. 7  0x0810d6f6 in main (argc=10, argv=0xbfffed54) at main.cpp:88
</pre>

**Comment by taskwarrior** [comment]
Migrated metadata:
```
Created: 2010-11-20T06:29:49Z
Modified: 2014-02-09T02:11:52Z
```

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2010-11-20T09:06:09Z says:_ 

 Confirmed.  Thank you for the stack dump.

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2010-11-20T09:43:29Z says:_ 

 Thank you Itay Perl - your name is now in the AUTHORS file by way of thanks.

Task was closing the data files before it had finished using them.  This only occurred for the 'log' command, and only if the task had a project specified.  The bug is fixed in 1.9.4, and should be released before end of year.  Here are the workarounds:

1. Instead of this command:
<pre>
$ task log <description> project:<project>
</pre>
Use these:
<pre>
$ task add <description> project:<project>
$ task <id> done
</pre>

2. If you are building task from source, apply this patch:
<pre>
diff --git a/src/command.cpp b/src/command.cpp
index 8ffa6b7..ebfd907 100644
* -- a/src/command.cpp
+++ b/src/command.cpp
@@ -195,12 +195,12 @@ int handleLog (std::string& outs)
     context.tdb.lock (context.config.getBoolean ("locking"));
     context.tdb.add (context.task);
     context.tdb.commit ();
*     context.tdb.unlock ();
 
     if (context.config.getBoolean ("echo.command"))
       out << "Logged task.\n";
 
     context.footnote (onProjectChange (context.task));
+    context.tdb.unlock ();
 
     outs = out.str ();
     context.hooks.trigger ("post-log-command");
</pre>

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2011-07-23T17:54:23Z says:_ 

 Note that this 1.9.4 bug was cherry-picked and back-ported to 1.9.3 for Ubuntu Natty release 11.04.
Thanks to Bryce Harrington.

https://bugs.launchpad.net/ubuntu/+source/task/+bug/733942
